jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/validation/BPMN2SyntaxChecker.java

```diff
@@ -322,20 +324,52 @@ private void checkFlowNode(FlowNode flowNode, FlowElementsContainer container, P
 	}
 
 	private void checkBusinessRuleTask(BusinessRuleTask businessRuleTask) {
-		Iterator<FeatureMap.Entry> biter = businessRuleTask.getAnyAttribute().iterator();
-		boolean foundRuleflowGroup = false;
-		while(biter.hasNext()) {
-			FeatureMap.Entry entry = biter.next();
-			if(entry.getEStructuralFeature().getName().equals("ruleFlowGroup")) {
-				foundRuleflowGroup = true;
-				String ruleflowGroup = (String) entry.getValue();
-				if(isEmpty(ruleflowGroup)) {
-					addError(businessRuleTask, new ValidationSyntaxError(businessRuleTask, BPMN2_TYPE, SyntaxCheckerErrors.BUSINESS_RULE_TASK_NO_RULEFLOW_GROUP));
+		String ruleLanguage = businessRuleTask.getImplementation();
+		if (RuleSetNode.DMN_LANG.equals(ruleLanguage)) {
+			ruleLanguage = BpmnMarshallerHelper.RULE_LANG_DMN;
+		} else {
+			ruleLanguage = BpmnMarshallerHelper.RULE_LANG_DRL;
+		}
+		if (ruleLanguage.equals(BpmnMarshallerHelper.RULE_LANG_DRL)) {
+			Iterator<FeatureMap.Entry> biter = businessRuleTask.getAnyAttribute().iterator();
+			boolean foundRuleflowGroup = false;
+			while (biter.hasNext()) {
+				FeatureMap.Entry entry = biter.next();
+				if (entry.getEStructuralFeature().getName().equals("ruleFlowGroup")) {
+					foundRuleflowGroup = true;
+					String ruleflowGroup = (String) entry.getValue();
+					if (isEmpty(ruleflowGroup)) {
+						addError(businessRuleTask, new ValidationSyntaxError(businessRuleTask, BPMN2_TYPE, SyntaxCheckerErrors.BUSINESS_RULE_TASK_NO_RULEFLOW_GROUP));
+					}
 				}
 			}
-		}
-		if(!foundRuleflowGroup) {
-			addError(businessRuleTask, new ValidationSyntaxError(businessRuleTask, BPMN2_TYPE, SyntaxCheckerErrors.BUSINESS_RULE_TASK_NO_RULEFLOW_GROUP));
+			if (!foundRuleflowGroup) {
+				addError(businessRuleTask, new ValidationSyntaxError(businessRuleTask, BPMN2_TYPE, SyntaxCheckerErrors.BUSINESS_RULE_TASK_NO_RULEFLOW_GROUP));
+			}
+		} else if (ruleLanguage.equals(BpmnMarshallerHelper.RULE_LANG_DMN)) {
+			boolean missingNamespace = true;
+			boolean missingModel = true;
+			if (businessRuleTask.getIoSpecification() != null) {
+				List<DataInput> dataInputs = businessRuleTask.getIoSpecification().getDataInputs();
+
+				if (dataInputs != null) {
+					for (DataInput di : dataInputs) {
+						if (di.getName().equals("namespace")) {
```

Comment:
Whole `else` block for DMN is pretty defensive and checks for null values. Are you sure that di.getName() returns always non null value? I am just asking.